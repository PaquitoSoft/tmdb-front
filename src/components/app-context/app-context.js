import React, { useContext } from 'react';

const AppContext = React.createContext({
	apiClient: undefined
});

export function AppProvider({ apiClient, children } ) {
	return (
		<AppContext.Provider value={{ apiClient }}>
			{children}
		</AppContext.Provider>
	);
}

export function useAppContext() {
	const context = useContext(AppContext);
	if (!context) {
		throw new Error('useAppContext must be used within an AppContext.Provider')
	}
	return context;
}
