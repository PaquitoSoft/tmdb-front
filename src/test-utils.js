import React from 'react';
import { shallow, mount } from 'enzyme';
import { AppProvider } from './components/app-context/app-context';

const userIdMock = 'hskfjahshjdjskla';

const imagesConfigMock = {
	"baseUrl":"https://image.tmdb.org/t/p/",
	"backdropSizes":["w300","w780","w1280","original"],
	"logoSizes":["w45","w92","w154","w185","w300","w500","original"],
	"posterSizes":["w92","w154","w185","w342","w500","w780","original"],
	"profileSizes":["w45","w185","h632","original"],
	"stillSizes":["w92","w185","w300","original"]
};

const renderers = {
	'shallow': shallow,
	'mount': mount
};

function AppContextMock({ children, apiClientMock }) {
	return (
		<AppProvider 
			apiClient={apiClientMock} 
			userId={userIdMock} 
			imagesConfig={imagesConfigMock}
		>
			{children}
		</AppProvider>
	);
}

export function render(
	component, 
	options = { renderType: 'shallow' }
) {
	return renderers[options.renderType](component);
}

export function validateSnapshot(
	component, 
	options = { renderType: 'shallow' }
) {
	const wrapper = render(component, options);
	expect(wrapper).toMatchSnapshot();
	return wrapper;
}

export function renderInAppContext(
	component, 
	options = {}
) {
	const _options = {
		renderType: 'mount',
		apiClientQueryMock: jest.fn().mockImplementation(() => Promise.resolve({})),
		apiClientMutationMock: jest.fn().mockImplementation(() => Promise.resolve({})),
		...options
	};

	const apiClientMock = {
		sendQuery: _options.apiClientQueryMock,
		sendMutation: _options.apiClientMutationMock
	};
	
	const wrapper = render(
		(
			<AppContextMock apiClientMock={apiClientMock}>
				{component}
			</AppContextMock>
		),
		_options
	);

	return wrapper.find(AppProvider).childAt(0);
}

export const clone = object => JSON.parse(JSON.stringify(object));
