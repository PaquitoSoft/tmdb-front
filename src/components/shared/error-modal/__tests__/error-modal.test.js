import React from 'react';
import { validateSnapshot, render } from '../../../../test-utils';
import * as AppContext from '../../../app-context/app-context';

import ErrorModal from '../error-modal';

describe('ErrorModal', () => {

	it('Should render error modal', () => {
		validateSnapshot(<ErrorModal error={new Error('Booom!')} />);
	});

	it('Should reset the error when clicking the Close button', () => {
		const setErrorMock = jest.fn();
		jest.spyOn(AppContext, 'useAppContext')
			.mockImplementation(() => ({	setError: setErrorMock }));
		
		const wrapper = render(<ErrorModal error={new Error('Booom!')} />);
		
		wrapper.find('.error-modal__close-button').simulate('click');
		expect(setErrorMock.mock.calls.length).toBe(1);
	});

});
