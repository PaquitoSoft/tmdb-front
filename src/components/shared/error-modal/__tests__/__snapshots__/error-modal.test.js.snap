// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal Should render error modal 1`] = `
<div
  className="error-modal"
>
  <div
    className="error-modal__veil"
  />
  <div
    className="error-modal__modal"
  >
    <div
      className="error-modal__head"
    />
    <div
      className="error-modal__content"
    >
      <p>
        Ouch! It has been a totally unexpected error. Please try again in a minute.
      </p>
    </div>
    <div
      className="error-modal__footer"
    >
      <button
        className="error-modal__close-button"
        onClick={[Function]}
      >
        Close
      </button>
    </div>
  </div>
</div>
`;
